addons/snailmail_account/wizard/account_move_send.py

```diff
@@ -77,16 +76,17 @@ def _hook_if_success(self, moves_data, from_cron=False, allow_fallback_pdf=False
         # EXTENDS 'account'
         super()._hook_if_success(moves_data, from_cron=from_cron, allow_fallback_pdf=allow_fallback_pdf)
 
-        moves = self.env['account.move']
-        for move, move_data in moves_data.items():
-            if move_data.get('send_by_post') and move.invoice_pdf_report_id:
-                moves |= move
-        if not moves:
-            return
-
-        self.env['snailmail.letter']\
-            .create([
-                self._prepare_snailmail_letter_values(move)
-                for move in moves
+        to_send = {
+            move: move_data
+            for move, move_data in moves_data.items()
+            if move_data.get('send_by_post') and move.invoice_pdf_report_id
+        }
+        if to_send:
+            self.env['snailmail.letter'].create([
+                {
+                    'user_id': self.env['res.partner'].browse(move_data['sp_partner_id']).user_id.id,
```

Comment:
This could be a performance issue. Backup the user id as well instead?

Reply:
Sure, but we're in stable, we might have pending invoices to send that don't have the newly `sp_user_id` set, what should be the fallback value ? :thinking:

Reply:
@clbr-odoo odoobot I guess.... we have no choice anyway :sweat_smile: